doc/build/orm/mapping_styles.rst

```diff
@@ -120,6 +120,86 @@ user-defined class, linked together with a :func:`.mapper`.  When we talk about
 "the behavior of :func:`.mapper`", this includes when using the Declarative system
 as well - it's still used, just behind the scenes.
 
+Mapping dataclasses and attrs
+-----------------------------
+
+The dataclasses_ module, added in Python 3.7, provides a ``dataclass`` class
+decorator to automatically generate boilerplate definitions of ``__init__()``,
+``__eq__()``, ``__repr()__``, etc. methods. Another very popular library that does
+the same, and much more, is attrs_. Classes defined using either of these can
+be mapped with the following caveats.
+
+.. note::
+
+    * Only classical mapping is possible, not Declarative. Classes inheriting from
```

Comment:
**mike bayer** (zzzeek) wrote:

this is not really accurate.  all mapper() calls in 2.0 will be using the declarative system, which has no reliance on metaclasses to do its work.  I'm adding new test cases now.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2157